Tests/test_file_jpeg.py

```diff
@@ -91,15 +91,17 @@ def test_cmyk(self):
             assert k > 0.9
 
     def test_dpi(self):
-        def test(xdpi, ydpi=None):
-            with Image.open(TEST_FILE) as im:
-                im = self.roundtrip(im, dpi=(xdpi, ydpi or xdpi))
-            return im.info.get("dpi")
-
-        assert test(72) == (72, 72)
-        assert test(300) == (300, 300)
-        assert test(100, 200) == (100, 200)
-        assert test(0) is None  # square pixels
+        for test_image_path in [TEST_FILE, "Tests/images/pil_sample_cmyk.jpg"]:
```

Comment:
I would suggest using [pytest.parameterize](https://docs.pytest.org/en/latest/parametrize.html) in future tests, helps #4193. In my experience, it is a lot easier to both read and debug. WDYT?

Reply:
Sure, and I've pushed a commit for this. It'll just take me a while to internalise that this is the new way of doing things.